packages/plugin-legacy/index.js

```diff
@@ -559,6 +565,44 @@ function replaceLegacyEnvBabelPlugin() {
   })
 }
 
+/**
+ * @param {any} targets
+ * @param {boolean} ignoreBrowserslistConfig
+ * @returns {'defaults'|undefined}
+ */
+function getDefaultTargets(targets, ignoreBrowserslistConfig) {
+  if (targets) return targets
+
+  if (ignoreBrowserslistConfig) {
+    return defaultTargets
+  }
+
+  try {
+    // fix https://github.com/vitejs/vite/issues/2476
+    // @babel/preset-env automatically detects browserslist config sources:
+    // browserslist field in package.json
+    // .browserslistrc file in cwd.
+
+    const getRealPath = (dir) => path.posix.join(process.cwd(), dir)
+    const pkgStr = fs.readFileSync(getRealPath('package.json'), 'utf8')
+    const pkgJson = JSON.parse(pkgStr)
+
+    // if the browserslist field in package.json
+    if (pkgJson.browserslist) {
+      return undefined
```

Comment:
**thought (non-blocking):** We may consider to return `null` instead of `undefined`
But only if this don't produce problems with the use of `targets` in line 35 (haven't looked into the whole file)